libsigner/src/events.rs

```diff
@@ -334,47 +342,86 @@ fn process_stackerdb_event(
     let event: StackerDBChunksEvent = serde_json::from_slice(body.as_bytes())
         .map_err(|e| EventError::Deserialize(format!("Could not decode body to JSON: {:?}", &e)))?;
 
-    let signer_event = if event.contract_id == boot_code_id(MINERS_NAME, is_mainnet) {
-        let blocks: Vec<NakamotoBlock> = event
-            .modified_slots
-            .iter()
-            .filter_map(|chunk| read_next::<NakamotoBlock, _>(&mut &chunk.data[..]).ok())
-            .collect();
-        SignerEvent::ProposedBlocks(blocks)
-    } else if event.contract_id.name.to_string().starts_with(SIGNERS_NAME)
-        && event.contract_id.issuer.1 == [0u8; 20]
-    {
-        let Some((signer_set, _)) =
-            get_signers_db_signer_set_message_id(event.contract_id.name.as_str())
-        else {
-            return Err(EventError::UnrecognizedStackerDBContract(event.contract_id));
-        };
-        // signer-XXX-YYY boot contract
-        let signer_messages: Vec<SignerMessage> = event
-            .modified_slots
-            .iter()
-            .filter_map(|chunk| read_next::<SignerMessage, _>(&mut &chunk.data[..]).ok())
-            .collect();
-        SignerEvent::SignerMessages(signer_set, signer_messages)
-    } else {
-        info!(
-            "[{:?}] next_event got event from an unexpected contract id {}, return OK so other side doesn't keep sending this",
-            local_addr,
-            event.contract_id
-        );
-        if let Err(e) = request.respond(HttpResponse::empty(200u16)) {
-            error!("Failed to respond to request: {:?}", &e);
+    let event_contract_id = event.contract_id.clone();
+
+    let signer_event = match SignerEvent::try_from(event) {
+        Err(e) => {
+            info!(
+                "[{:?}] next_event got event from an unexpected contract id {}, return OK so other side doesn't keep sending this",
+                local_addr,
+                event_contract_id
+            );
+            ack_dispatcher(request);
+            return Err(e.into());
         }
-        return Err(EventError::UnrecognizedStackerDBContract(event.contract_id));
+        Ok(x) => x,
     };
 
-    if let Err(e) = request.respond(HttpResponse::empty(200u16)) {
-        error!("Failed to respond to request: {:?}", &e);
-    }
+    ack_dispatcher(request);
 
     Ok(signer_event)
 }
 
+impl TryFrom<StackerDBChunksEvent> for SignerEvent {
+    type Error = EventError;
+
+    fn try_from(event: StackerDBChunksEvent) -> Result<Self, Self::Error> {
```

Comment:
I refactored this into a `TryFrom` implementation because the miner's coordinator doesn't use the RPC callback interface. Instead, it receives the StackerDB events directly over a synchronous channel.